torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1229,6 +1225,25 @@ def test_all_reduce_result_cuda(self):
                 self.assertEqual(result, [_build_tensor(src + 1, expected_value)])
             self._barrier()
 
+        def call_dist_op(self, profiling_title, async_op, op, *args, expect_event=True, **kwargs):
+            with torch.autograd.profiler.profile() as prof:
+                work = op(*args, async_op=async_op, **kwargs)
+                if async_op:
+                    work.wait()
+                    work._get_profiling_future().wait()
```

Comment:
It looks like `_get_profiling_future` is only used here? If so, why do we need this? Once we exit the the profiler context, the profiling information should be recorded appropriately.

Reply:
I don't think that's necessarily the case, as `work.wait()` could return without the profiling callback having ran and this wait ensures that the profiling callback (one that terminates the record function)is finished successfully. We had to do something similar with RPC (see: https://github.com/pytorch/pytorch/pull/38352/files) although in that case it is transparent to the user.

Ideally when there's profiling, similar to RPC, work.wait() should ensure the profiling callbacks have ran before returning. It might depend on future/work merge, though maybe we can get it to work now by modifying `::wait()` to await the profiling future if one exists